src/Composer/Installer.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\ManagerPlugin\Composer;
+
+use Composer\Package\Locker;
+use Contao\ManagerPlugin\Dependency\DependencyResolverTrait;
+use Contao\ManagerPlugin\Dependency\DependentPluginInterface;
+
+class Installer
+{
+    use DependencyResolverTrait;
+
+    /**
+     * @var string
+     */
+    private static $generatedClassTemplate = <<<'PHP'
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\ManagerPlugin;
+
+use Contao\ManagerPlugin\Bundle\BundlePluginInterface;
+use Contao\ManagerPlugin\Config\ConfigPluginInterface;
+use Contao\ManagerPlugin\Config\ExtensionPluginInterface;
+use Contao\ManagerPlugin\Routing\RoutingPluginInterface;
+
+/**
+ * This class has been auto-generated. It will be overwritten at every run of
+ * `composer install` or `composer update`.
+ * 
+ * @see \Contao\ManagerBundle\Composer\Installer
+ */
+%s
+{
+    public const BUNDLE_PLUGINS = BundlePluginInterface::class;
+    public const CONFIG_PLUGINS = ConfigPluginInterface::class;
+    public const EXTENSION_PLUGINS = ExtensionPluginInterface::class;
+    public const ROUTING_PLUGINS = RoutingPluginInterface::class;
+
+    /**
+     * @var array
+     */
+    private $plugins;
+
+    /**
+     * @var array
+     */
+    private $disabled = [];
+
+    public function __construct()
+    {
+        $this->plugins = %s;
+    }
+
+    /**
+     * Returns all active plugin instances.
+     *
+     * @return array
+     */
+    public function getInstances()
+    {
+        return array_diff_key($this->plugins, array_flip($this->disabled));
+    }
+
+    /**
+     * Returns the active plugin instances of a given type (see class constants).
+     *
+     * @param string $type
+     * @param bool   $reverseOrder
+     *
+     * @return array
+     */
+    public function getInstancesOf($type, $reverseOrder = false)
```

Comment:
Since the minimum requirement is PHP 7.1, we should make this `string $type, $reverseOrder = false): array`

Reply:
done in b2ebae71102129721fe5765c6c1dd34af40e9b21